code-challenges/2d-arrays/challenges-12.test.js

```diff
@@ -0,0 +1,318 @@
+    store: 'Pet store',
+    items: [{ name: 'Cans of food', quantity: 8 }, { name: 'Treats', quantity: 24 }, { name: 'Leash', quantity: 1 }]
+  }
+];
+
+const howManyTreats = (arr) => {
+  // Solution code here...
+  return arr[2].items[1].quantity;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named battleship that accepts a 2D array and two numbers: a row coordinate and a column coordinate.
+
+Return "hit" or "miss" depending on if there's part of a boat at that position in the array. Assume the array has only one of two values at each index. '#' for part of a boat, or ' ' for open water.
+
+Here is a sample board:
+[
+  ['#', ' ', '#', ' '],
+  ['#', ' ', '#', ' '],
+  ['#', ' ', ' ', ' '],
+  [' ', ' ', '#', '#'],
+]
+
+The top row of the board is considered row zero and row numbers increase as they go down.
+------------------------------------------------------------------------------------------------ */
+
+const battleship = (board, row, col) => {
+  //  Solution code here...
+  return board[row][col] === '#' ? 'hit' : 'miss';
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named calculateProduct that takes in a two-dimensional array of numbers, multiplies all of the numbers in each array, and returns the final product. This function should work for any number of inner arrays.
+
+For example, the following input returns a product of 720: [[1,2], [3,4], [5,6]]
+------------------------------------------------------------------------------------------------ */
+
+const calculateProduct = (numbers) => {
+  // Solution code here...
+  let productCalculated = 1;
+  for (let i = 0; i < numbers.length; i++) {
+    for (let j = 0; j < numbers[i].length; j++)
+      productCalculated *= numbers[i][j];
+  }
+  return productCalculated;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 6
+
+Write a function named averageDailyTemperature that accepts a two-dimensional array representing average daily temperatures grouped week-by-week.
+
+Calculate the average daily temperature during that entire period. Your output should be a single number. Write your function so it could accept an array with any number of weeks given to it.
+------------------------------------------------------------------------------------------------ */
+
+// Real daily average temperatures for Seattle, October 1-28 2017
+const weeklyTemperatures = [
+  [66, 64, 58, 65, 71, 57, 60],
+  [57, 65, 65, 70, 72, 65, 51],
+  [55, 54, 60, 53, 59, 57, 61],
+  [65, 56, 55, 52, 55, 62, 57],
+];
+
+const averageDailyTemperature = (weather) => {
+  // Solution code here...
+  let amountOfDays = 0;
+  let temperatures = 0;
+  for (let i = 0; i < weather.length; i++) {
+    for (let j = 0; j < weather[i].length; j++) {
+      amountOfDays++;
```

Comment:
Outside of this inner loop you could have had mount of days += weather[i].length so you didn't have to increment amountOfDays for every single day (your solution is totally valid though!!!)